src/huggingface_hub/hf_api.py

```diff
@@ -492,7 +495,9 @@ def logout(self, token: Optional[str] = None) -> None:
             token (``str``, `optional`):
                 Hugging Face token. Will default to the locally saved token if not provided.
         """
-        logging.error("This method is deprecated in favor of `unset_access_token`.")
+        warnings.warn(
```

Comment:
These warnings are not tested I think. Could you please add tests, similar to the one here: 


https://github.com/huggingface/huggingface_hub/pull/733/files#diff-5e02bb367c95034caad81bbd7a52410b9a05cb29878494288f74ef7805822679R153-R172

Reply:
Thanks for the link, added!